lib/Infer/ExhaustiveSynthesis.cpp

```diff
@@ -86,6 +88,60 @@ namespace {
 // experiment with synthesizing at reduced bitwidth, then expanding the result
 // aggressively avoid calling into the solver
 
+using CostFunctionType = std::function<float(Inst *, bool)>;
+
+float defaultCostFunction(Inst* I, bool IgnoreDepsWithExternalUses) {
+  return souper::cost(I, IgnoreDepsWithExternalUses);
+}
+
+// TODO : Derive these randomly to maximize infer rate with test-infer.sh
```

Comment:
what does "derive these randomly" mean?

Reply:
Pick a plausible cost range for each instruction and sample frequency.
For all possible cost combinations, run test-infer.sh on the solver/alive directory and pick the combinations with the best infer success rate.

Reply:
don't explain it here, explain it in the code